workflow/controller/operator.go

```diff
@@ -351,18 +351,23 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.log.WithError(err).Error("error in entry template execution")
 		// we wrap this error up to report a clear message
 		x := fmt.Errorf("error in entry template execution: %w", err)
+		shouldDeletePVCs := true
+
 		switch err {
 		case ErrDeadlineExceeded:
 			woc.eventRecorder.Event(woc.wf, apiv1.EventTypeWarning, "WorkflowTimedOut", x.Error())
 		case ErrParallelismReached:
+			shouldDeletePVCs = false
```

Comment:
Instead of this `true` `false` logic can you just throw an error when the error is `ErrParallelismReached`